lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -19,19 +19,58 @@ Fp16_to_Fp8E5M2(Location loc, ConversionPatternRewriter &rewriter,
   fp16x2Vec1 = insert_element(fp16x2VecTy, fp16x2Vec1, v[2], i32_val(0));
   fp16x2Vec1 = insert_element(fp16x2VecTy, fp16x2Vec1, v[3], i32_val(1));
 
+  Value a0 = bitcast(fp16x2Vec0, i32_ty);
+  Value a1 = bitcast(fp16x2Vec1, i32_ty);
+  
+  auto fp8x4VecTy = vec_ty(i8_ty, 4);
+  a0 = bitcast(a0, fp8x4VecTy); 
+  a1 = bitcast(a1, fp8x4VecTy); 
+
+  return {extract_element(i8_ty, a0, i32_val(1)),
+	  extract_element(i8_ty, a0, i32_val(3)),
+	  extract_element(i8_ty, a1, i32_val(1)),
+	  extract_element(i8_ty, a1, i32_val(3))
+	  };
+}
+#else
+const std::string Fp16_to_Fp8E5M2 =
+    "{                            \n"
+    ".reg .b32 a<2>;              \n"
+    "and.b32 a0, $1, 0xfffefffe;  \n"   // a0 &= 0xfffefffe
+    "and.b32 a1, $2, 0xfffefffe;  \n"   // (strip lowest bit)
+    "add.u32 a0, a0, 0x00800080;  \n"   // a0 += 0x00800080
+    "add.u32 a1, a1, 0x00800080;  \n"   // (round to nearest)
+    "prmt.b32 $0, a0, a1, 0x7531; \n\t" // output = a1a0
+    "}";
+#endif
+
+#ifdef USE_ROCM
+static SmallVector<Value>
+Fp16_to_Fp8E5M2FNUZ(Location loc, ConversionPatternRewriter &rewriter,
```

Comment:
Shall we just use the native conversion instructions on MI300 (so does NV for e4m3 on Hopper)? The SW conversion doe s not handle denorm properly.

Reply:
Yes, we should. I will do that in another PR, reason is the changes are bigger than what I had thought. We need to differentiate the software/hw instr conversion on MI200 and MI300, which need to change the function signature.

Is that OK for you?

Reply:
We can do it in another PR. I don't feel that we need to change the function signature though. Can we use guards like "ifdef __gfx940__" to differentiate here? Or the counterpart of the computeCapability attribute on the ROCm side?

Reply:
Since we figured that MI2xx GPUs do not need to support AMD fp8 datatypes, does it mean we do not need sw conversions for these new types?
Or do we still need it since the native conversion instructions on MI300 can only convert f32 but not f16? So we'll need f8 --> f32 --> f16 and f16 --> f32 --> f8?